include/libvfio-user.h

```diff
@@ -550,6 +590,18 @@ vfu_dma_write(vfu_ctx_t *vfu_ctx, dma_sg_t *sg, void *data);
 
 /*
  * Supported PCI regions.
+ *
+ * Note to developers: in VFIO, each region starts at a terabyte offset
+ * (VFIO_PCI_INDEX_TO_OFFSET) and because Linux supports up to 128 TB of
+ * virtual memory, there can be up to 128 device regions. PCI regions are fixed
+ * and in retrospect this choice has proven to be problematic because devices
+ * might contain potentially unused regions. New regions can now be positioned
+ * anywhere by using the VFIO_REGION_INFO_CAP_TYPE capability.
+ * In vfio-user we don't have this problem because the region index is just an
+ * identifier: the VMM memory maps a file descriptor that is passed to it and
+ * the offset that it memory maps is relative to the beginning of the region,
```

Comment:
I think the crucial point to make here is that the mapping offset is derived from mmap_areas offset value, rather than a static mapping from region index to offset.

Reply:
Thanks, I know the way I wrote it was a bit convoluted and was hoping in a suggestion :).